src/jaybenne/sourcing.cpp

```diff
@@ -31,8 +31,21 @@ TaskStatus SourcePhotons(T *md, const Real t_start, const Real dt) {
   auto pm = md->GetParentPointer();
   auto &resolved_pkgs = pm->resolved_packages;
   auto &jb_pkg = pm->packages.Get("jaybenne");
+  const Real h = jb_pkg->template Param<Real>("planck_constant");
   auto &eos = jb_pkg->template Param<EOS>("eos_d");
-  auto &opacity = jb_pkg->template Param<Opacity>("opacity_d");
+  int n_nubins = -1;
+  Real numin = -1.;
+  Real numax = -1.;
+  MeanOpacity mopacity;
+  Opacity opacity;
+  if constexpr (FT == FrequencyType::gray) {
+    mopacity = jb_pkg->template Param<MeanOpacity>("mopacity_d");
+  } else if constexpr (FT == FrequencyType::multigroup) {
+    opacity = jb_pkg->template Param<Opacity>("opacity_d");
+    n_nubins = jb_pkg->template Param<int>("n_nubins");
+    numin = jb_pkg->template Param<Real>("numin");
+    numax = jb_pkg->template Param<Real>("numax");
+  }
```

Comment:
Maybe just a comment for posterity, but since `n_nubins`, `numin`, and `numax` are "poisoned" to invalid values above, should there be assertions for valid values after this block?

Reply:
I'm torn on this point. On the one hand, assertion checks like you say would be extra safety, but I think code can get cluttered when it becomes mostly DBC checks. And the poisoned values should be pretty obvious down the line... What the right balance of those two considerations is for the jaybenne project is for you and others to decide :)

Reply:
Great point. This seems like a sensible enough argument to not add them to me.